### PR TITLE
Parametric observer

### PR DESCRIPTION
**Pull request by juloberno**
- parametric observer implementation

### CODE REVIEW

File: bark/models/observer/observer_model_parametric.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+
+ObservedWorld ObserverModelParametric::Observe(
+    const WorldPtr& world, const AgentId& agent_id) {
+  
+  // Clone world here since otherwise we change global world state
+  auto observed_world = ObservedWorld(world->Clone(), agent_id);
+
+  AddStateDeviationFrenet(observed_world.GetEgoAgent(), ego_state_deviation_dist_);
+  for (auto& agent : observed_world.GetOtherAgents()) {
```
Comment (patrickhart) [inline bark/models/observer/observer_model_parametric.cpp:48]: do we only want the possibility of the other agents to be perturbed or should we also introduce it for the ego-agent, e.g. modeling a poor localization?

File: bark/models/observer/observer_model_parametric.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  current_frenet_state.lat += frenet_deviation[1];
+  current_frenet_state.vlat += frenet_deviation[2];
+  current_frenet_state.vlon += frenet_deviation[3];
+  current_frenet_state.angle += frenet_deviation[4];
+  
+  // Convert back and set dynamic agent state 
+  const auto deviated_state = FrenetStateToDynamicState(current_frenet_state, lane_corridor->GetCenterLine());
+  agent->SetCurrentState(deviated_state);
+
+  return;
```
Comment (patrickhart) [inline bark/models/observer/observer_model_parametric.cpp:76]: is a return in a void really required? better to return a boolean instead maybe?

File: bark/models/observer/observer_model_parametric.hpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+using bark::commons::DistributionPtr;
+
+/**
+ * @brief  Observer creating an ObservedWorld given a World
+ * @note   
+ * @retval None
+ */
+class ObserverModelParametric : public ObserverModel {
+ public:
+  explicit ObserverModelParametric(bark::commons::ParamsPtr params);
```
Comment (patrickhart) [inline bark/models/observer/observer_model_parametric.hpp:45]: add `using bark::commons::ParamsPtr` above and pass as const ref: `const ParamsPtr& params`

File: bark/world/objects/agent.hpp [hunk 4]
```diff
@@ -170,6 +170,10 @@ class Agent : public Object {
     history_ = history;
   }
 
+  void SetCurrentState(const State& current_state) {
+    history_.back().first = current_state;
```
Comment (patrickhart) [inline bark/world/objects/agent.hpp:174]: should we not fill the action also with an default value?

**Review by patrickhart — APPROVED** [review]

File: bark/models/observer/observer_model_parametric.cpp (hunk 1 above)
Comment (juloberno) [reply bark/models/observer/observer_model_parametric.cpp:48]: See line 47, where this is implemented

File: bark/models/observer/observer_model_parametric.cpp (hunk 2 above)
Comment (juloberno) [reply bark/models/observer/observer_model_parametric.cpp:76]: I check the conventions

File: bark/world/objects/agent.hpp (hunk 4 above)
Comment (juloberno) [reply bark/world/objects/agent.hpp:174]: Yeah, thats right there is missing the case of the history being empty. But the function is only called after the constructor where the first entry of the history is always set

File: bark/models/observer/observer_model_parametric.hpp (hunk 3 above)
Comment (juloberno) [reply bark/models/observer/observer_model_parametric.hpp:45]: okay

File: bark/world/objects/agent.hpp (hunk 4 above)
Comment (juloberno) [reply bark/world/objects/agent.hpp:174]: Yeah, thats right there is missing the case of the history being empty. But the function is only called after the constructor where the first entry of the history is always set

**Pull request merged**